### PR TITLE
consul: able to set pass/fail thresholds on consul service checks

### PR DESCRIPTION
**Pull request by shoenig**
This change adds the ability to set the fields `success_before_passing` and
`failures_before_critical` on Consul service check definitions. This is a
feature added to Consul v1.7.0 and later.
  https://www.consul.io/docs/agent/checks#success-failures-before-passing-critical

Nomad doesn't do much besides pass the fields through to Consul.

### CODE REVIEW

File: nomad/structs/services.go [hunk 1]
```diff
@@ -257,6 +267,14 @@ func (sc *ServiceCheck) validate() error {
 		}
 	}
 
+	if sc.SuccessBeforePassing < 0 {
```
Comment (tgross) [inline nomad/structs/services.go:270]: The [Consul docs](https://www.consul.io/docs/agent/checks.html#success-failures-before-passing-critical
) say:
> This feature is available for HTTP, TCP, gRPC, Docker & Monitor checks. 

so we should probably validate that the check isn't a script check here?

File: nomad/structs/services.go [hunk 2]
```diff
@@ -309,6 +327,10 @@ func (sc *ServiceCheck) Hash(serviceID string) string {
 	// use name "true" to maintain ID stability
 	hashBool(h, sc.GRPCUseTLS, "true")
 
+	// Only include pass/fail if non-zero to maintain ID stability with Nomad < 0.12
```
Comment (tgross) [inline nomad/structs/services.go:330]: :+1: nice catch

File: website/pages/docs/job-specification/service.mdx [hunk 3]
```diff
@@ -202,6 +202,12 @@ scripts.
   service. Valid options are the empty string, `passing`, `warning`, and
   `critical`.
 
+- `success_before_passing` `(int:0)` - The number of successful checks required before
```
Comment (drewbailey) [inline website/pages/docs/job-specification/service.mdx:205]:
```suggestion
- `success_before_passing` `(int:0)` - The number of consecutive successful checks required before
```

It's consecutive right? So 0 is just one successful check?

File: nomad/structs/services.go (hunk 1 above)
Comment (shoenig) [reply nomad/structs/services.go:274]: good idea, added!

File: website/pages/docs/job-specification/service.mdx (hunk 3 above)
Comment (shoenig) [reply website/pages/docs/job-specification/service.mdx]: Ah yep, fixed

**Pull request merged**